### PR TITLE
Add no-std feature

### PR DESCRIPTION
**Pull request by hujw77**

### CODE REVIEW

File: src/std.rs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#[cfg(feature = "std")]
+pub use std::{borrow, cmp, fmt, format, time, vec, vec::Vec};
+
+#[cfg(not(feature = "std"))]
+pub use alloc::{borrow, format, string, vec, vec::Vec};
```
Comment (acgxv) [inline src/std.rs:5]: why not `core::borrow`

File: src/test_utils/mod.rs [hunk 2]
```diff
@@ -50,7 +50,9 @@ pub fn create_rng() -> RAND {
 // Reads the json test files
 pub fn json_reader(file_name: &str) -> BufReader<File> {
     let mut file_path_buf = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
-    file_path_buf.push("src/test_utils/hash_to_curve_vectors/");
+    file_path_buf.push("src");
```
Comment (acgxv) [inline src/test_utils/mod.rs:53]: any reasons?

**Review by acgxv — APPROVED** [review]

File: src/test_utils/mod.rs (hunk 2 above)
Comment (hujw77) [reply src/test_utils/mod.rs:53]: This commit is from upstream, maybe cause by one version late sync.

File: src/std.rs (hunk 1 above)
Comment (hujw77) [reply src/std.rs:5]: Cause of `ToOwned`

**Pull request merged**